### DISCUSSION TITLE
Image upload with rest api - jwt and header

### DISCUSSION THREAD

**Discussion by oferardi** [AxisCommunications/discussions#248] [ACAP]
Hi
I need to upload still image to a bucket of appwrite with rest api.

Is it possible to implement it with developing acap application? And if not, is it possible with acap using an external library?

Thanks

POST /v1/storage/buckets/{bucketId}/files HTTP/1.1 Host: HOSTNAME Content-Type: multipart/form-data; boundary="cec8e8123c05ba25" X-Appwrite-Response-Format: 1.0.0 X-Appwrite-Project: 5df5acd0d48c2 X-Appwrite-JWT: eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ... Content-Length: *Length of your entity body in bytes* --cec8e8123c05ba25 Content-Disposition: form-data; name="fileId" "[FILE_ID]" --cec8e8123c05ba25 Content-Disposition: form-data; name="file" cf 94 84 24 8d c4 91 10 0f dc 54 26 6c 8e 4b bc e8 ee 55 94 29 e7 94 89 19 26 28 01 26 29 3f 16... --cec8e8123c05ba25 Content-Disposition: form-data; name="permissions[]" ["read(\"any\")"] --cec8e8123c05ba25--

**Comment by mattias-kindborg-at-work** [comment]
Hi @oferardi.

Please provide more context to your problem, as defined in the [guidelines](https://github.com/orgs/AxisCommunications/discussions/13). 

There's always more people asking questions than there are people answering them. That creates a balancing act. If you minimize the investment and effort it takes for someone to understand your question by e.g. giving them the background to your problem, what alternatives you've considered, what limitations you're affected by, links to the APIs you mention, camera models you intend to target. I guess there's also a JWT token in play here, which you failed to mention, that has to be understood by parsing the request headers.

Questions without adequate information will be left unanswered.

Edit 1: I now see that JWT is mentioned in the title, but JWT is a pretty complicated topic and you need to provide more context for someone to answer this. The structure of the JWT token, and the flow which it is created, has to be detailed in a complex questions like this one.
